feat(cdp): add ecommerce events spec (#10210)
PostHog/posthog.com: contents/docs/data/event-spec/ecommerce-events.mdx

---
title: Ecommerce events specification
sidebar: Docs
showTitle: true
---

PostHog's ecommerce spec helps define the journey for a customer as they browse your store, click on promotions, view products, add those products to a cart, and complete a purchase.

> 

Author: ```suggestion

This spec serves as the foundation for our CDP destination templates. We recommend implementing this spec to avoid having to manually map events and properties to our CDP destinations.

```

---

> title: Ecommerce Events Specification

Reviewer: ```suggestion
title: Ecommerce events specification
```

sidebar: Docs
showTitle: true
---

> PostHog's e-commerce spec helps define the journey for a customer as they browse your store, click on promotions, view products, add those products to a cart, and complete a purchase.

Reviewer: Can you provide a bit more context here? 
- Where is this used in PostHog? 
- We recommend people capture these events to best work with certain features of PostHog?
- Do any of the libraries autocapture these events?

```suggestion
PostHog's ecommerce spec helps define the journey for a customer as they browse your store, click on promotions, view products, add those products to a cart, and complete a purchase.
```

Author: > Where is this used in PostHog?
 
We will base our cdp destination templates on top of this event spec.

> We recommend people capture these events to best work with certain features of PostHog?

Yes, following this event spec, will allow you to easily spin up cdp destinations without having to manually map the correct events or properties to each destination.

> Do any of the libraries autocapture these events?

None of our current SDKs autocapture these or other e-commerce events.

Reviewer: Perfect, thanks!

## Event lifecycle overview

Here is a list of supported events for our various categories within the customer journey.

### Browsing

| Event                  | Description                              |
|------------------------|------------------------------------------|
| Products Searched      | User searched for products               |
| Product List Viewed    | User viewed a product list or category   |
| Product List Filtered	 | User filtered a product list or category |

### Promotions

| Event              | Description               |
|--------------------|---------------------------|
| Promotion Viewed   | User viewed promotion     |

> | Promotion Clicked	 | User clicked on promotion |
> 

Reviewer: ```suggestion
| Promotion Clicked  | User clicked on promotion |
```

| share_via     | String | Method of sharing          |
| share_message | String | Message sent with the cart |
| recipient     | String | Recipient of the share     |
| cart_id       | String | Cart ID                    |
| products      | Array  | Products in the cart       |

Example:

```js
posthog.capture('Cart Shared', {
  share_via: 'email',
  share_message: 'Hey, check out this item',
  recipient: 'max@posthog.com',
  cart_id: 'e1f17b1ed3ec47a298a05838',
  products: [
    { product_id: 'dc2046c33b104932851dbcf4' },
    { product_id: '17584e91ae1abff0d355e3f5' }
  ]
});
```

## Reviewing

Track customer reviews of products.

### Product Reviewed

Fire this event when a product is reviewed.

| Property    | Type   | Description                |
|-------------|--------|----------------------------|
| product_id  | String | Database ID of the product |
| review_id   | String | Review ID                  |
| review_body | String | Review content             |
| rating      | String | Review rating              |

Example:

```js
posthog.capture('Product Reviewed', {
  product_id: 'dc2046c33b104932851dbcf4',
  review_id: 'e6b795d687884d0b8b59b4de',

>   review_body: 'They're shipping things, like, almost every day. That's so great.',

Reviewer: ```suggestion
  review_body: 'They are shipping things, like, almost every day. That is so great.',
```
